app/main/posts/posts-routes.js

```diff
@@ -1,155 +1,158 @@
 module.exports = [
     '$stateProvider',
+    '$urlMatcherFactoryProvider',
 function (
-    $stateProvider
+    $stateProvider,
+    $urlMatcherFactoryProvider
 ) {
+    $urlMatcherFactoryProvider.strictMode(false);
 
     $stateProvider
     .state(
         {
-            name: 'list',
+            name: 'list', /** I agree, this is a terrible name for this state. Ideas are welcome.**/
             abstract: true,
             controller: require('./views/post-views.controller.js'),
             template: require('./views/main.html'),
+            params: {
+                view: {value: null, squash: true},
+                filterState: {value: null, squash: true},
+                savedSearchState: {value: null, squash: true}
+            },
             resolve: {
                 isLoading: function () {
                     return {state: true};
+                },
+                collection: ['$transition$', 'CollectionEndpoint', function ($transition$, CollectionEndpoint) {
+                    if ($transition$.params().collectionId) {
+                        return CollectionEndpoint.get({collectionId: $transition$.params().collectionId}).$promise;
```

Comment:
What happens if these resolve params fail?

Reply:
so if the params fail, meaning you don't have a collectionId, nothing happens except you won't have a collection. Now, if the id is there in transition$.params().collectionId but it fails, I think I would need an error handler to take you to a 404.. right? it's something we didn't ever have (right now, if you go to a post that doesn't exist in any version of the web app, you get an empty view.. :( ) . 

I can handle that today

Reply:
TLDR: If a resolve returns a rejected promise, the transition WILL FAIL miserably . Need to handle that